tests/e2e/containers/config.go

```diff
@@ -40,22 +40,25 @@ const (
 // Returns ImageConfig needed for running e2e test.
 // If isUpgrade is true, returns images for running the upgrade
 // Otherwise, returns images for running non-upgrade e2e tests.
-func NewImageConfig(isUpgrade bool) ImageConfig {
+func NewImageConfig(isUpgrade, isFork bool) ImageConfig {
 	config := ImageConfig{
 		RelayerRepository: relayerRepository,
 		RelayerTag:        relayerTag,
 	}
 
-	if isUpgrade {
+	if isUpgrade && !isFork {
```

Comment:
Was playing with this and adding comments to make sure I understand. Feel free to adapt as you see fit.

```
    if !isUpgrade {
	// If upgrade is not tested, we do not need InitRepository and InitTag
	// because we directly call the initialization logic without
	// the need for Docker.
        config.OsmosisRepository = CurrentBranchOsmoRepository
	config.OsmosisTag = CurrentBranchOsmoTag
        return isUpgrade
    }

    // If upgrade is tested, we need to utilize InitRepository and InitTag
    // to initialize older state with Docker
    config.InitRepository = previousVersionInitRepository
    config.InitTag = previousVersionInitTag

    if isFork {
	// Forks are state compatible with earlier versions before fork height.
	// Normally, validators switch the binaries pre-fork height
	// Then, once the fork height is reached, the state breaking-logic
	// is run.
	config.OsmosisRepository = CurrentBranchOsmoRepository
	config.OsmosisTag = CurrentBranchOsmoTag
    } else {
	// Upgrades are run at the time when upgrade height is reached
	// and are submitted via a governance proposal. Thefore, we
	// must start running the previous Osmosis version. Then, the node
        // should auto-upgrade, at which point we can restart the updated
        // Osmosis validator container.
        config.OsmosisRepository = previousVersionOsmoRepository
	config.OsmosisTag = previousVersionOsmoTag
    }
```